cisco_asa/icon_cisco_asa/util/api.py

```diff
@@ -0,0 +1,83 @@
+import json
+import requests
+from collections import OrderedDict
+from insightconnect_plugin_runtime.exceptions import PluginException
+
+
+class CiscoAsaAPI:
+    def __init__(self, username: str, password: str, url: str, verify_ssl: bool, port: int, logger: object):
+        self.url = url + ":" + str(port) + "/api/"
+        self.verify_ssl = verify_ssl
+        self.logger = logger
+        self.username = username
+        self.password = password
+
+    def get_groups(self):
+        return self.run_with_pages("objects/networkobjectgroups")
+
+    def get_group(self, group_name: str):
+        for item in self.get_groups():
+            if item.get("name") == group_name or item.get("objectId") == group_name:
+                return item
+
+        return {}
+
+    def get_objects(self):
+        return self.run_with_pages("objects/networkobjects")
+
+    def run_with_pages(self, path):
+        objects = []
+        limit = 100
+        for page in range(0, 9999):
+            response = self._call_api(
+                "GET",
+                path,
+                params={
+                    "limit": limit,
+                    "offset": page * limit
+                }
+            )
+            objects.extend(response.get("items", []))
+
+            if (page + 1) * limit >= response.get("rangeInfo", {}).get("total", 0):
+                break
+
+        return objects
+
+    def get_clock(self):
+        return self._call_api("GET", "monitoring/clock")
+
+    def _call_api(self, method: str, path: str, json_data: dict = None, params: dict = None):
+        response = {"text": ""}
+        headers = OrderedDict([
+            ('Content-Type', 'application/json'),
+            ('User-Agent', 'test-user-agent')
```

Comment:
This is just an example of what our instance has - a user's agent will be different. The default should be `REST API Agent` but allow the user to specify a custom one in the connection - see connection feedback